http_server/atoi_test.py

```diff
@@ -0,0 +1,64 @@
+from amaranth.sim import Simulator
+
+from atoi import AtoI
+
+dut = AtoI(32)
+
+async def run_test_case(ctx, input: str, expected: int) -> int: 
+    ctx.set(dut.reset, 1)
+    await ctx.tick()
+    ctx.set(dut.reset, 0)
+
+    for c in input:
+        ctx.set(dut.input.payload, ord(c))
+        ctx.set(dut.input.valid, 1)
+        await ctx.tick()
+        ctx.set(dut.input.valid, 0)

```

Comment:
May be worth having a test where we tick on an invalid cycle too (eh, that's a later problem)